### PR TITLE
added the event schedule, and bootswatch theme

### PR DESCRIPTION
**Pull request by clintBock**
rip me up niko, i wanted to clean it up but niko wants to rip me

### CODE REVIEW

File: Zenith/Controllers/EventsController.cs [hunk 1]
```diff
@@ -22,7 +24,57 @@ public class EventsController : Controller
             var events = db.Events.Include(E => E.ActivityCategory);
             return View(await events.ToListAsync());
         }
+        public async Task<ActionResult> Schedule()
```
Comment (nikojohnarellano) [inline Zenith/Controllers/EventsController.cs]: async is not needed for this method there's no await happening inside

File: Zenith/Controllers/EventsController.cs [hunk 2]
```diff
@@ -22,7 +24,57 @@ public class EventsController : Controller
...
+              .Select(i => startOfWeek
+                 .AddDays(i)
+                 .ToString("dd-MMMM-yyyy")));
+            var weekDays = result.Split(',');
+            List<DateTime> dates = new List<DateTime>();
+            foreach(var w in weekDays)
+            {
+                dates.Add(Convert.ToDateTime(w));
+            }
+
```
Comment (nikojohnarellano) [inline Zenith/Controllers/EventsController.cs]: The way you're making the dates of the week is too complicated. You can replace lines 37-47 in one line basically:

`List<DateTime> list = Enumerable.Range(0, 7).Select(i => startOfWeek.AddDays(i)).ToList<DateTime>();`

File: Zenith/Controllers/EventsController.cs [hunk 3]
```diff
@@ -22,7 +24,57 @@ public class EventsController : Controller
...
+            
+
+            return new ScheduleViewModel()
+            {
+                DaysAndEvents = dic
+            };
 
+
+        }
+        public string GetDateString(DateTime date)
```
Comment (nikojohnarellano) [inline Zenith/Controllers/EventsController.cs]: You dont need this method. You can get this format of a DateTime object by just doing: \

`date.ToString("D")`

File: Zenith/Controllers/EventsController.cs [hunk 4]
```diff
@@ -22,7 +24,57 @@ public class EventsController : Controller
...
+            }
+
+            //can't get this this to work; can't use datetimes in linq query, all online says is dbfunctions.truncatetime but still doesn't work
+            //var events = db.Events.Include(e => e.ActivityCategory).Where(e => dates.Any(d => DbFunctions.TruncateTime(d.Date) == e.CreationDate.Date)).ToList();
+
+            var events = db.Events.Include(e => e.ActivityCategory).ToList();
+            Dictionary<string, List<Event>> dic = new Dictionary<string, List<Event>>();
+            foreach (var d in dates)
+            {
+                dic.Add(GetDateString(d), new List<Event>());
```
Comment (nikojohnarellano) [inline Zenith/Controllers/EventsController.cs]: No need for a call to GetDateString(). Just do:

`d.ToString("D")`

File: Zenith/Views/Events/Schedule.cshtml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+                foreach (var e in day.Value)
+                {
+                    <tr>
+                        <td>@e.StartDateTime.TimeOfDay - @e.EndDateTime.TimeOfDay</td>
+                        <td>@e.ActivityCategory.ActivityDescription</td>
+                    </tr>
+                }
+
+            }
+        }
```
Comment (nikojohnarellano) [inline Zenith/Views/Events/Schedule.cshtml]: According to the instructions, the Schedule table needs to be shown in the Home Page. So we'll eventually need to move this

File: Zenith/Controllers/EventsController.cs [hunk 6]
```diff
@@ -22,7 +24,57 @@ public class EventsController : Controller
             var events = db.Events.Include(E => E.ActivityCategory);
             return View(await events.ToListAsync());
         }
+        public async Task<ActionResult> Schedule()
+        {
+            return View(getDic());
+        }
+        public ScheduleViewModel getDic()
```
Comment (nikojohnarellano) [inline Zenith/Controllers/EventsController.cs]: Minor but method name should start with a capital letter to be consistent looool

**Review by nikojohnarellano — APPROVED** [review]

**Review by itsmattli — APPROVED** [review]
Pretty good, about exactly what I'd expect out of the Lead Senior Developer of Clickpics

**Comment by clintBock** [comment]
nice

**Pull request merged**